src/synthetix/StakingRewards.sol

```diff
@@ -162,11 +163,14 @@ contract StakingRewards is IStakingRewards, Pausable, ReentrancyGuard {
         emit Recovered(tokenAddress, tokenAmount);
     }
 
-    function setRewardsDuration(uint256 _rewardsDuration) external onlyOwner {
-        require(
-            block.timestamp > periodFinish,
-            "Previous rewards period must be complete before changing the duration for the new period"
-        );
+    function setRewardsDuration(uint256 _rewardsDuration) external onlyOwner updateReward(address(0)) {
+        uint256 periodFinish_ = periodFinish;
+        if (block.timestamp < periodFinish_) {
+            uint256 leftover = (periodFinish_ - block.timestamp) * rewardRate;
```

Comment:
`periodFinish_ - block.timestamp` could potentially be in an `unchecked { }` block, but I'm not sure the extra stack item wouldn't offset the gains in terms of gas.
Just flagging, but it's probably better to leave this as it is.

Reply:
Prefer to leave as-is to match the code in `notifyRewardAmount`  which has a similar subtraction that was not updated either during the upgrade to solidity 0.8. Besides, `setRewardsDuration` is not expected to be frequently called.